ibis/expr/operations.py

```diff
@@ -2746,6 +2746,27 @@ def output_type(self):
         return rules.shape_like(self.args[0], map_type.value_type)
 
 
+class MapValueOrDefaultForKey(ValueOp):
+
+    input_type = [
+        rules.map(dt.any, dt.any),
+        rules.one_of((dt.string, dt.int_), name='key'),
+        rules.value(name='default')
+    ]
+
+    def output_type(self):
+        map_type = self.args[0].type()
+        value_type = map_type.value_type
+        default_type = self.args[2].type()
+
+        if not (default_type is dt.null or
```

Comment:
Shouldn't the default value for `default` be `NULL`

Reply:
good point

Reply:
pushed, I don't know how to make default optional though, ``rules.value`` doesn't like this